src/components/FlowRunGraphArtifactsPopover.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <FlowRunGraphPopover
+    v-if="artifactsSelection"
+    class="flow-run-graph-artifacts-popover"
+    :position="artifactsSelection.position"
+    @on-close="close"
+  >
+    <h4 class="flow-run-graph-artifacts-popover__label">
+      Artifacts
+    </h4>
+    <div class="flow-run-graph-artifacts-popover__content">
+      <template v-for="artifactId in artifactsSelection.ids" :key="artifactId">
+        <FlowRunGraphArtifactCard :artifact-id="artifactId" />
+      </template>
+    </div>
+  </FlowRunGraphPopover>
+</template>
+
+<script lang="ts" setup>
+  import { GraphItemSelection, isArtifactsSelection } from '@prefecthq/graphs'
+  import { computed } from 'vue'
+  import { FlowRunGraphPopover, FlowRunGraphArtifactCard } from '@/components'
+
+  const props = defineProps<{
+    selection: GraphItemSelection | null,
+  }>()
+
+  const emit = defineEmits<{
+    'update:selection': [null],
+  }>()
+
+  const artifactsSelection = computed(() => {
+    if (props.selection && isArtifactsSelection(props.selection)) {
+      return props.selection
+    }
+
+    return null
+  })
```

Comment:
Same thought here about this component only accepting an ArtifactsSelection rather than a GraphItemSelection. 

You might want a wrapper component that accepts a selection | null and then mounts the appropriate popover component depending on what type of selection there is.